lib/puppet/face/config.rb

```diff
@@ -71,7 +81,18 @@
                      _("New environment loaders generated from the requested section.")) do
         # And now we can lookup values that include those from environments configured from
         # the requested section
+
+        if @default_section
+          Puppet.warning (_(<<-EOM) % { section_name: options[:section] } ).chomp
```

Comment:
These should be separate translation sentences.

messages = []
messages << _("No section specified; defaulting to '%{section_name}'.") % { section_name: options[:section] }
#TRANSLATORS '--section' is a command line option and should not be translated
messages << _("Set the config section by using the `--section` flag.")
#TRANSLATORS `puppet config --section user print foo` is a command line example and should not be translated
messages << _("For example, `puppet config --section user print foo`.")
messages << _("For more information, see https://puppet.com/docs/puppet/latest/configuration.html")
Puppet.warning(messages.join('\n'))

or if you want them to be one line

Puppet.warning(messages.join(' '))

Reply:
@ericldelaney updated, and I also gave a similar treatment to the HEREDOC I got this message from.